### PR TITLE
feat: add entities:configure command

### PR DESCRIPTION
**Pull request by grof**

### CODE REVIEW

File: src/commands/entities/configure.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+  outputHumanReadable(transformedData: any) {
+    debug('outputHumanReadable()')
+    const {entityId, name} = transformedData
+    this.log(`Entity ${chalk.cyan(name)} with ID ${chalk.cyan(entityId)} was updated.`)
+  }
+
+  setRequestActionMessage(options: any) {
+    debug('setRequestActionMessage()')
+    this.requestActionMessage = `Configuring entity ${options.name} in project ${options.project}`
```
Comment (oliver-daniel) [inline src/commands/entities/configure.ts:144]: `options.project` is a numerical ID here, right?

File: src/mix/api/entities.ts [hunk 2]
```diff
@@ -22,6 +23,25 @@ import {MixClient, MixResponse} from '../types'
...
 
+/**
+ * Configure a new entity in a project.
+ *
+ * @category entities
+ */
+export async function configureEntity(client: MixClient, params: EntitiesConfigureParams): Promise<MixResponse> {
+  debug('configureEntity()')
+  const {entityName, projectId, ...bodyParams} = params
+  const body = buildCreateOrUpdateEntityBody(bodyParams)
```
Comment (oliver-daniel) [inline src/mix/api/entities.ts:34]: I'm not sure how I feel about a function with `or` in its name. Would it be messier/more confusing to abstract this method name to something more generic -- maybe with an underscore -- then have both `buildCreateEntityBody` and `buildUpdateEntityBody` return calls to it?

File: src/commands/entities/configure.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+const debug = makeDebug('mix:commands:entities:configure')
+
+export default class EntitiesConfigure extends MixCommand {
+  static description = `configure an entity
+
+Use this command to configure an entity in a project.
+
+NOTE: This command cannot change the type of an entity.
+When using this command, you must provide the current type
+of the entity. Use command entities:convert to change the
```
Comment (oliver-daniel) [inline src/commands/entities/configure.ts:29]: Should we include this line ahead of `entities:convert` existing? Or is it coming out in the same release as this PR

Comment (oliver-daniel) [inline src/commands/entities/configure.ts:29]: What a UX pain. Not our 'fault', of course, but it's quite irritating that users need to supply a type just to get the command to work. I'm almost tempted to suggest an `--entity-type=auto` mode, which GETs the project behind the scenes to secure the type.

File: src/commands/entities/configure.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+locale. The --pattern and --locale flags matter only to entities
+of type "regex".
+
+Relationial entities can have zero or one isA relation and
+zero or many hasA relations. One --is-A or --has-A flag must be
+set at a minimum for an entity of type "relational".
+
+The examples below show how to configure each type of entity.
+In each example, every allowed flag is used.
+Note that if a value is not provided for a particular flag,
```
Comment (oliver-daniel) [inline src/commands/entities/configure.ts:48]: What does this mean?

**Review by oliver-daniel — COMMENTED** [review]
As promised, this command is quite similar to other `entities` commands already merged in. I left a handful of questions, mostly about semantics, documentation and the like, but on the whole it looks good.

File: src/commands/entities/configure.ts (hunk 3 above)
Comment (grof) [reply src/commands/entities/configure.ts:29]: Yes, that would indeed require a second endpoint call and also bring up other usability issues. For instance, there would need to be error handling around trying to update a base entity, which is something you cannot get to currently as --entity-type does not accept 'base' as a possible value. We are going to proceed with having users provide the --entity-type flag but took note of it.

File: src/mix/api/entities.ts (hunk 2 above)
Comment (grof) [reply src/mix/api/entities.ts:34]: As discussed, I introduced an internal `buildEntityPayload` function that gets called by the two you suggested.

**Review by oliver-daniel — APPROVED** [review]
Looks good.

**Pull request merged**